### PR TITLE
br: modify collate.newCollationEnabled according to the config of the cluster

### LINKED ISSUE #39150: Restore meet error: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create a cluster with tidb config `new_collations_enabled_on_first_bootstrap: false`.
2. create a table with collation 'utf8mb4_0900_ai_ci': `create table `books` (`id` bigint auto_increment, `name` varchar(64), primary key (`id`)) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;`
3. insert several records to the table.
4. execute a full backup
5. create another cluster tidb config `new_collations_enabled_on_first_bootstrap: false`.
6. restore the backup to the new cluster

### 2. What did you expect to see? (Required)

restore successfully

### 3. What did you see instead (Required)

restore failed with error `Error: [ddl:1273]Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'`

### 4. What is your TiDB version? (Required)

v6.1.0

### PR DESCRIPTION
**Pull request by MoCuishle28**
### What problem does this PR solve?

Issue Number: close #39150

Problem Summary:
The `collate.newCollationEnabled` variable is set to 1 when the collate package is initialized. When using this package, this value is not modified according to the config of the TiDB cluster (i.e., `new_collations_enabled_on_first_bootstrap`).

As a result, the user sets `new_collations_enabled_on_first_bootstrap` to `false` in the TiDB cluster, and it is still treated as `true` when br is executed.

### What is changed and how it works?
Modify `collate.newCollationEnabled` according to the config of the cluster.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Manual test
- create a cluster with tidb config `new_collations_enabled_on_first_bootstrap: false`.
- create a table with collation 'utf8mb4_0900_ai_ci': `create table books (id bigint auto_increment,name varchar(64), primary key (id)) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci`;
- insert several records to the table.
- execute a full backup
- create another cluster tidb config `new_collations_enabled_on_first_bootstrap: false`.
- restore the backup to the new cluster => success

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix a bug that: "Error Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'" even if new_collation_enabled is false.
```

### CODE REVIEW

**Comment by MoCuishle28** [comment]
Due to the cycle import caused by importing the `br/pkg/gluetidb` package when writing unit tests, we moved the `CheckNewCollationEnable` func to the `br/pkg/restore` package and moved the const `tidbNewCollationEnabled` to the `br/pkg/utils` package.

**Review by WangLe1321 — APPROVED** [review]
LGTM

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -2495,3 +2496,50 @@ func TidyOldSchemas(sr *stream.SchemasReplace) *backup.Schemas {
...
+
+	// collate.newCollationEnabled is set to 1 when the collate package is initialized,
+	// so we need to modify this value according to the config of the cluster
+	// before using the collate package.
+	enabled := false
+	if newCollationEnable == "True" {
+		enabled = true
+	}
+	// modify collate.newCollationEnabled according to the config of the cluster
+	collate.SetNewCollationEnabledForTest(enabled)
```
Comment (3pointer) [inline br/pkg/restore/client.go:2542]: this function seems used for test code? Is there any other better way to set it?

Comment (3pointer) [inline br/pkg/restore/client.go:2542]: this function seems used for test code? Is there any other better way to set it?

Comment (MoCuishle28) [reply br/pkg/restore/client.go:2542]: This is also used in addition to testing.
e.g., https://github.com/pingcap/tidb/blob/48e17e2ecb3fbc987c77b77c094c26f37f1dc9fb/session/session.go#L2959
and https://github.com/pingcap/tidb/blob/48e17e2ecb3fbc987c77b77c094c26f37f1dc9fb/util/dbutil/table.go#L34
Seems this is the only function that modifies `newCollationEnabled`.

**Review by 3pointer — APPROVED** [review]

File: br/pkg/restore/client.go [hunk 2]
```diff
@@ -2495,3 +2496,50 @@ func TidyOldSchemas(sr *stream.SchemasReplace) *backup.Schemas {
...
+			backupNewCollationEnable, newCollationEnable)
+	}
+
+	// collate.newCollationEnabled is set to 1 when the collate package is initialized,
+	// so we need to modify this value according to the config of the cluster
+	// before using the collate package.
+	enabled := false
+	if newCollationEnable == "True" {
+		enabled = true
+	}
```
Comment (YuJuncen) [inline br/pkg/restore/client.go:2540]: Perhaps register the boolean values directly?
```suggestion
	enabled := newCollationEnable == "True" 
```

**Review by YuJuncen — APPROVED** [review]
rest lgtm

**Pull request merged**